Libraries/LibWeb/WebAssembly/Memory.cpp

```diff
@@ -83,10 +94,32 @@ WebIDL::ExceptionOr<void> Memory::reset_the_memory_buffer()
     auto& vm = this->vm();
     auto& realm = *vm.current_realm();
 
-    MUST(JS::detach_array_buffer(vm, *m_buffer, JS::PrimitiveString::create(vm, "WebAssembly.Memory"_string)));
-
-    auto buffer = TRY(create_a_memory_buffer(vm, realm, m_address));
-    m_buffer = buffer;
+    // FIXME: 1. Let map be the surrounding agent's associated Memory object cache.
+    // FIXME: 2. Assert: map[memaddr] exists.
+    // FIXME: 3. Let memory be map[memaddr].
+    // FIXME: 4. Let buffer be memory.[[BufferObject]].
+    // 5. If IsFixedLengthArrayBuffer(buffer) is true,
+    if (m_buffer->is_fixed_length()) {
+        // https://webassembly.github.io/threads/js-api/index.html#refresh-the-memory-buffer
```

Comment:
We don't have this proposal implemented, why are we linking to/following its js-api?

Reply:
Ah, the shared buffer support is from that proposal.
that's alright then, not sure if this code will be run given that we don't actually support that proposal though?

Reply:
It'll be run in some capacity because WPT uses some of these APIs to explicitly get a handle to the SharedArrayBuffer constructor, even when the spec says it's not supposed to visible 🙃

Reply:
Maybe wpt can be pointed to the spec and told "back off bud"?

Reply:
Unlikely, it doesn't look like anyone cares about fixing https://github.com/whatwg/html/issues/5380 when you can do this: https://github.com/web-platform-tests/wpt/blob/7b6f075f0ec3c52d92c61fb9f35ce0ec927b17b2/common/sab.js#L3

Reply:
While implementing `threading` proposal is definitely an interesting task, I'm afraid it's a bit too complicated for me at the moment. This PR is a step to make some of WPT tests to work (ones that use `WebAssembly.Memory` to created shared array buffer). 

To my understanding, there is nothing to be changed in this PR in the scope of this conversation, so I resolve it, as per the contribution guide.